src/main/java/com/mongodb/jdbc/MongoDatabaseMetaData.java

```diff
@@ -406,19 +406,38 @@ public ResultSet getTableTypes() throws SQLException {
 
     // Helper for getting a stream of all database names.
     private Stream<String> getDatabaseNames() {
-        return this.conn.mongoClient.listDatabaseNames().into(new ArrayList<>()).stream();
+        return this.conn
+                .mongoClient
+                .listDatabaseNames()
+                .into(new ArrayList<>())
+                .stream()
+                .filter(dbName -> !dbName.isEmpty());
+    }
+
+    // Helper for getting a list of collection names from the db
+    // Using runCommand instead of listCollections as listCollections does not support authorizedCollections option
+    private ArrayList<Document> getCollectionsFromRunCommand(MongoDatabase db) {
+        Document listCollectionsResult =
+                db.runCommand(
```

Comment:
I am also wondering if we could use this runCommand : <TResult> TResult runCommand(Bson command, Class<TResult> resultClass) with TResult being MongoIterable<MongoListTablesResult> or ListCollectionsIterable<MongoListTablesResult>.
This way you don't need the new constructor for MongoListTablesResult and we don't have a special deserialization process for this and it looks like we could have a proper batch cursor and even use iterators out of the box.

Reply:
I added the `MongoRunCmdListTablesResult ` class for this.  I looked into `ListCollectionsIterable` and `MongoIterable` but could not find way to make them work.

Reply:
Yep, I looked into it a bit too.
I'm pretty sure there is a way, but it does not seem trivial because ListCollectionsOperation does not have the ability to let us set authorizedCollections. We would have to extends a few classes to get this to work.
I am wondering why this is not a problem for Java driver users.
However, I am still concern that we will discard Collections if there is more than 1 batch of data.
Can you ask the Java driver team or check other tools (mongosh, other drivers) to see how they deal with this?

Reply:
We don't have to worry about batches. The command response is a bit quirky, but unless Server changes something, all collections will be returned under `firstBatch`. [Here is the relevant code](https://github.com/mongodb/mongo/blob/master/src/mongo/db/commands/list_collections.cpp#L255-L261).

Reply:
Thanks Nathan for helping clarify that and providing the relevant code. 👍